__tests__/server/utils/pollModuleMap.spec.js

```diff
@@ -350,22 +263,34 @@ describe('pollModuleMap', () => {
     loadModulesPromise = Promise.reject(error);
     await pollModuleMap();
     expect(loadModules).toHaveBeenCalledTimes(1);
+    expect(console.error).toHaveBeenCalledTimes(1);
+    expect(console.error).toHaveBeenCalledWith('pollModuleMap: error polling', error);
+
+    expect(setTimeout).toHaveBeenCalledTimes(1);
+    expect(setTimeout).toHaveBeenCalledWith(pollModuleMap, 5e3);
 
-    return loadModulesPromise
-      // catch and then so as to run after pollModuleMap's error handling
-      .catch((err) => err)
-      .then(() => {
-        expect(console.error).toHaveBeenCalledTimes(1);
-        expect(console.error).toHaveBeenCalledWith('pollModuleMap: error polling', error);
-
-        expect(setTimeout).toHaveBeenCalledTimes(1);
-        expect(setTimeout).toHaveBeenCalledWith(pollModuleMap, 5e3);
-
-        expect(setGauge).toHaveBeenCalledTimes(1);
-        expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
-        expect(incrementGauge).toHaveBeenCalledTimes(1);
-        expect(incrementGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollConsecutiveErrors);
-      });
+    expect(setGauge).toHaveBeenCalledTimes(1);
+    expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
+    expect(incrementGauge).toHaveBeenCalledTimes(1);
+    expect(incrementGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollConsecutiveErrors);
+  });
+
+  it('resets the time to the next polling to the minimum if there were rejected modules', async () => {
+    const { default: pollModuleMap } = load();
+    loadModulesPromise = Promise.resolve({ rejectedModules: { 'bad-module': { reasonForRejection: 'not compatible' } } });
+    await pollModuleMap();
+    expect(loadModules).toHaveBeenCalledTimes(1);
+
+    expect(console.warn).toHaveBeenCalledTimes(1);
+    expect(console.warn).toHaveBeenCalledWith('pollModuleMap: 1 modules rejected:', ['bad-module: not compatible']);
+
+    expect(setTimeout).toHaveBeenCalledTimes(1);
+    expect(setTimeout).toHaveBeenCalledWith(pollModuleMap, 5e3);
+
+    expect(setGauge).toHaveBeenCalledTimes(1);
+    expect(setGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollWait, 5);
+    expect(incrementGauge).toHaveBeenCalledTimes(1);
+    expect(incrementGauge).toHaveBeenCalledWith(holocronMetrics.moduleMapPollConsecutiveErrors);
```

Comment:
Can we also check that `resetPollTime` was called

Reply:
https://github.com/americanexpress/one-app/pull/962/files#diff-82cec72196769195fc18f858b882683ecae08a9f5fc3b7d097fae4f60d33b404R288